### ISSUE TITLE
root executables :: add functions argument to the main root executable

### ISSUE THREAD

**Issue by adriansev** [root-project/root#8447]
### Explain what you would like to see improved
It is just a point of view but IMHO, it would be easier if all additional functionality (hadd, mkdir, cp, ls, etc...)
would be usable as arguments/verbs to root: `root -hadd ...`, `root -ls ...` 
Moreover `--` token for end of global arguments would be useful in the following way:
* if verb is present (cp, ls, mv, hadd , etc...) whatever is after `--` are arguments for the verb (otherwise the verb should be the last)
* if there is no verb present whatever is after `--` is text to be passed to root as code (replacement/addition to -e)

this can become even more useful if these verbs are also converted to special commands (like `.pwd`, `.ls`)
and if the line tokenization is also done for them (with the usual token of `;`)

**Comment by eguiraud** [comment]
Personally I think it makes sense that we have different commands for different jobs -- `root` is not a swiss-army knife program, it's a C++ interpreter with ROOT-specific features, kind of like `python`.

**Comment by adriansev** [comment]
@eguiraud while i agree with you, my thinking actually was also along having a common code base and also "a la python" usage: while there is the option of having an executable doing "import module, run main function" one can also do "python -m module module_args". so, my proposal was from a point of view of code (common code base, maybe also usable programatically in macros, as part maybe of some generic ROOT toolbox) and the second point of view was of actual usage where maybe `root -verb verb_args` could be easier and clearer to use than `root[verb] args` .. but in the end my proposal was more on the lines of "wouldn't you (devs) be interested in such a usage/point of view ?" 
and getting back to python: yes, it is exactly (also) a swiss-army knife like python :) and beside some bash script that i converted to python, i started to see a point of/play with having root scripts (with root in shebang ;) )

**Comment by Axel-Naumann** [comment]
Just FYI, we already have `-e expr` which allows to do some of what you're suggesting here. I will leave this open for a while to see whether I am mis-estimating and our user community actually wants this. If so, folks, please share your argument / use case for having this. If there no significant support shows up we will instead continue to invest in more fundamental parts of ROOT.

**Comment by Axel-Naumann** [comment]
We discussed this in the team and the conclusion is that you raised a really important point: how can people discover the binaries without such a functionality flag?

We decided against such a wrapper - mainly because it would add yet another layer of complexity (a python script then then forwards arguments etc). Instead, we will make sure that all ROOT binaries exist also as a version starting with `root`:  `rootls` `root-hadd` etc, such that they are way easier to discover for instance when installed in `/usr/local/bin` next to 100s of other binaries.

Thanks for suggesting this - well, something similar!
